float32 radius
float32 angle
float32[4] covariance
float32[2] start
float32[2] end
